angular/projects/public-lng/src/app/project/authorizations/authorizations.component.ts

```diff
@@ -25,12 +15,70 @@ export class AuthorizationsComponent implements OnInit {
   public id: string;
   public text: string[];
 
-  constructor(private dataService: DataService, private route: ActivatedRoute) {
+  public showpanel = true;
+  public filters = [{
+    'displayName': 'Government Agency',
+    'textFilters': [
+      {
+        'displayName': 'Environmental Assessment Office',
+        'fieldName': '_master.issuingAgency'
+      },
+      {
+        'displayName': 'BC Oil and Gas Commission',
+        'fieldName': '_master.issuingAgency'
+      }
+    ]
+  }];
+
+  constructor(
+    private dataService: DataService,
+    private route: ActivatedRoute,
+    private router: Router
+    ) {
     this.route.parent.params.subscribe(params => {
       this.id = params.id;
       this.text = this.dataService.getText(this.id, this.pageType);
     });
   }
 
   ngOnInit() {}
+
+  togglePanel() {
+    this.showpanel = !this.showpanel;
+  }
+
+  filterChange(event) {
+    // Generate new route keeping old params
+    // tslint:disable-next-line: prefer-const
+    let newParams = {};
+
+    // save default set of params, tack on new ones.
+    this.route.params.subscribe(params => {
+
+      // Filter out the incoming params (remove them entirely)
+      Object.keys(params).forEach(p => {
+        if (Object.keys(event).includes(p)) {
+          // We will be overriding this param later.
+        } else {
+          // Existing param we should save.
+          newParams[p] = params[p];
+        }
+      });
+
+      Object.keys(event).forEach(item => {
+        if (event[item] === undefined || event[item].length === 0) {
+          // console.log('skipping:', e);
+        } else {
+          newParams[item] = event[item];
+        }
```

Comment:
Instead of having an empty if(), you could do:

```
If (event[item] !== undefined && event[item] !== null && event[item].length > 0 {
newParams[item] = event[item]
```

Reply:
It's part comment and part left for debugging.  I don't think it hurts anything.